docs/proposals.rst

```diff
@@ -0,0 +1,112 @@
+.. proposals:
+
+Proposals
+=========
```

Comment:
I think we should have a `docs/proposals/` directory, with a proposal-per-file. This will make collaboration easier (if more than one proposal is being worked on at a time, you'll get needless conflicts)